pkg/controller/migmigration/stage.go

```diff
@@ -0,0 +1,415 @@
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+func (l *StagePodList) contains(pod StagePod) bool {
+	for _, srcPod := range *l {
+		if pod.equals(srcPod) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (l *StagePodList) merge(list ...StagePod) {
+	for _, pod := range list {
+		if !l.contains(pod) {
+			*l = append(*l, pod)
+		}
+	}
+}
+
+func (l *StagePodList) create(client k8sclient.Client) (int, error) {
+	existingPods := StagePodList{}
+	if len(*l) > 0 {
+		existingPods.list(client, (*l)[0].Labels)
+	}
+	counter := 0
+	for _, stagePod := range *l {
+		if existingPods.contains(stagePod) {
+			continue
+		}
+		err := client.Create(context.TODO(), &stagePod.Pod)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return 0, err
+		}
+		counter++
+	}
+
+	return counter + len(existingPods), nil
+}
+
+func (l *StagePodList) list(client k8sclient.Client, labels map[string]string) error {
+	podList := corev1.PodList{}
+	options := k8sclient.MatchingLabels(labels)
+	err := client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return err
+	}
+	for _, pod := range podList.Items {
+		l.merge(buildStagePodFromPod(migref.ObjectKey(&pod), pod.Labels, &pod))
+	}
+
+	return nil
+}
+
+func (t *Task) ensureStagePodsFromOrphanedPVCs() error {
+	existingStagePods := StagePodList{}
+	stagePods := StagePodList{}
+	client, err := t.getSourceClient()
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	err = existingStagePods.list(client, t.stagePodLabels())
+	if err != nil {
+		log.Trace(err)
+		return nil
+	}
+
+	pvcMounted := func(list StagePodList, claimName string) bool {
+		for _, pod := range list {
+			for _, volume := range pod.Spec.Volumes {
+				claim := volume.PersistentVolumeClaim
+				if claim != nil && claim.ClaimName == claimName {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	for _, ns := range t.sourceNamespaces() {
+		list := &corev1.PersistentVolumeClaimList{}
+		err = client.List(context.TODO(), k8sclient.InNamespace(ns), list)
+		if err != nil {
+			log.Trace(err)
+			return nil
+		}
+		for _, pvc := range list.Items {
+			t.getPVs()
+			// Exclude unbound PVCs
+			if pvc.Status.Phase != corev1.ClaimBound {
+				continue
+			}
+			if pvcMounted(existingStagePods, pvc.Name) {
+				continue
+			}
+			stagePods.merge(buildStagePod(pvc, t.stagePodLabels()))
+		}
+	}
+
+	created, err := stagePods.create(client)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if created > 0 {
+		t.Owner.Status.SetCondition(migapi.Condition{
+			Type:     StagePodsCreated,
+			Status:   True,
+			Reason:   Created,
+			Category: Advisory,
+			Message:  "[] Stage pods created.",
+			Items:    []string{strconv.Itoa(created)},
+			Durable:  true,
+		})
+	}
+
+	return nil
+}
+
+// Ensure all stage pods from running pods withing the application were created
+func (t *Task) ensureStagePodsFromTemplates() error {
```

Comment:
I may be missing something here, but there doesn't seem to be anything in `ensureStagePodsFromTemplates` that filters out the stage pods created from `ensureStagePodsFromRunning` -- will this result in two stage pods created here?

Reply:
bYeah, it is filtering those during `stagePods.create` call by comparing existing pods for equality in the volume.VolumeSource fields on line https://github.com/konveyor/mig-controller/pull/485/files#diff-656fd52107ab09d74773450b8b7d546bR77